lib/awesome_print/formatter.rb

```diff
@@ -107,18 +107,15 @@ def awesome_dir(d)
     # Utility methods.
     #------------------------------------------------------------------------------
 
-    # A class (ex. `Net::HTTP.Get`) might have `attr_reader :method` accessor
-    # which causes `object.method(:to_hash)` throw `ArgumentError (wrong number
-    # of arguments (given 1, expected 0))`. The following tries to avoid that.
-    def has_method_accessor?(object)
-      !object.method(:method)
-    rescue ArgumentError
-      true
-    end
+    KERNEL_METHOD = Kernel.instance_method(:method)
 
     def convert_to_hash(object)
-      return nil if has_method_accessor?(object)
-      return nil if !object.respond_to?(:to_hash) || object.method(:to_hash).arity != 0
+      return nil unless object.respond_to?(:to_hash)
+
+      # A class (ex. `Net::HTTP.Get`) might have `attr_reader :method` accessor
+      # which causes `object.method(:to_hash)` throw `ArgumentError (wrong number
+      # of arguments (given 1, expected 0))`. The following tries to avoid that.
+      return nil if KERNEL_METHOD.bind_call(object, :to_hash).arity != 0
```

Comment:
`UnboundMethod#bind_call` was added in Ruby 2.7. For backwards compatibility, I'll need to switch to:

```suggestion
      return nil if KERNEL_METHOD.bind(object).call(:to_hash).arity != 0
```